scope/forms.py

```diff
@@ -95,34 +71,29 @@ class Meta:
         )
 
     def clean_password(self):
-        data = self.cleaned_data["password"]
-        if (
-            self.cleaned_data["password"] != ""
-            and len(self.cleaned_data["password"]) < 8
-        ):
+        password = self.cleaned_data.get("password", "")
+        if password != "" and len(password) < 8:
```

Comment:
Thanks @cole!

That's right. The main "problem" in here is the intention to allow password changes (or not) in the same form when editing users. By default Django provides two different forms to do so, which may have been a better choice, but I didn't want to change the existing functionality.

The default UserChangeForm doesn't include the password fields and the way they were added was causing some translations and validations issues as you mentioned in here. Instead of changing those fields, I've found a way to unify both forms in one based on the UserCreationForm, which allows to use the validation and translation for those fields in both actions.

I've also moved the group processing to the form but I didn't have time to do the same with the user checks to determine the fields in the form.

Please, if you could take another look, I'll appreciate it.

Reply:
👍